internal/database/gitserver_repos.go

```
@@ -319,13 +320,62 @@ WHERE r.name = %s
 	return scanSingleGitserverRepo(s.QueryRow(ctx, sqlf.Sprintf(q, name)))
 }
 
-func scanSingleGitserverRepo(row *sql.Row) (*types.GitserverRepo, error) {
-	if row.Err() != nil {
-		return nil, errors.Wrap(row.Err(), "getting GitserverRepo")
+func (s *gitserverRepoStore) GetByNames(ctx context.Context, names ...api.RepoName) ([]*types.GitserverRepo, error) {
```

Where will this be used, is there a chance we'll hit the param limit?

---

it will be used for discovery of gitserver instance addresses. depending on a time-slot which will be used for names accumulation and then execution of the batch. I don't really believe that we will get any closer to the limit, but I always prefer a sane error message instead of an unexpected error from postgres. 

what do you think?

---

Yeah, totally agree we should return an error instead of passing it down to the database but it'll still end up being an error at runtime. 

Can we maybe use batching if needed?

---

It's very unlikely that we'd have 32k+ simultaneous gitserver client calls, but I don't think it hurts to have some batching here, especially if we decide to use this method in other situations.

---

my proposal is to merge it now as is, so we at least have this function in main, and then I will add batching in next commit.